proto/notify.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+service Notify {
+  rpc GetUser (ForUserRequest) returns (User) {}
```

Comment:
Each `rpc` should have their input as the name of the RPC plus `Request` and their output + `Response` even if it's empty for now e.g.:

```
rpc GetUser (GetUserRequest) returns (GetUserResponse) {}
```

E.g. see https://gophers.slack.com/archives/C8T4Y2248/p1533921988000302?thread_ts=1533920932.000058&cid=C8T4Y2248

> Good protobuf practice is to return something, even an empty DeleteResponse{}. The reasoning being a change from empty to a message is a breaking change, but a change to add a field is a backwards compatible change.
> Having done many years with pb in production environments, I've learned the hard way to follow that advice. :slightly_smiling_face:

Reply:
that post doesn't say anything about naming conventions. all of these rpc calls have a request/return message

Reply:
```
rpc GetUsers (Empty) returns (Users) {}
```

If `Empty` is used by any other `rpc`, then this is a problem. E.g. now you want to add some fields to the `GetUsers` call like `bool hide_inactive`, `int64 limit`. To do that they have to be added to `Empty` which doesn't make sense if other calls are also using `Empty`. So you then change the input to `GetUsersRequest` and now have a breaking change. So it's better to just use the Request/Response rule from the beginning even if it's a tiny bit more upfront work.

For the record, I had a quick look in GCP's repo and they have some cases where they use the "pragmatic" way like you have done:

```
rpc GetService (ServiceRequest) returns (Service) {}
```

However, you can't add anything to `Service` without affecting other things, so it's safest to just use `GetServiceRequest` and `GetServiceResponse`.

Reply:
this makes sense